doc/source/check_api_doc.py

```diff
@@ -0,0 +1,112 @@
+from docutils.parsers.rst import directives
+import docutils.nodes
+
+sys.path.insert(0, os.path.abspath('../../'))
+import larray as la
+import larray.random as random
+
+
+_exclude = ['Group', 'OrderedDict', 'basestring', 'x', 'is_interactive_interpreter', 'float_error_handler_factory']
+
+
+def get_items_from_larray():
+    def check_not_deprecated(value):
+        if not isinstance(value, types.FunctionType):
+            return True
+        return not 'renamed_to' in value.__qualname__
+
+    future_funcs = dir(sys.modules['__future__'])
+    _larray_items = {key: value for key, value in vars(la).items()
+                    if not (key.startswith('_') or key in future_funcs or isinstance(value, (types.ModuleType)))}
+    _larray_items.update({'random': random})
+
+    larray_items = {}
+    deprecated_items = []
+    for key, value in _larray_items.items():
+        if key not in _exclude:
+            # if module
+            if isinstance(value, types.ModuleType):
+                module_name = key
+                larray_items[module_name] = 'Module'
+                func_and_constants = {k: v for k, v in vars(value).items() if not k.startswith('_')}
+                for k, v in func_and_constants.items():
+                    if check_not_deprecated(v):
+                        api_name = '{}.{}'.format(module_name, k)
+                        larray_items[api_name] = 'Function' if isinstance(value, types.FunctionType) else 'Constant'
+                    else:
+                        deprecated_items.append(k)
+            # if class
+            elif isinstance(value, type):
+                class_name = key
+                larray_items[class_name] = 'Class'
+                methods_and_attrs = [k for k in dir(value) if not k.startswith('_')]
+                for k in methods_and_attrs:
+                    v = getattr(value, k)
+                    if check_not_deprecated(v):
+                        api_name = '{}.{}'.format(class_name, k)
+                        larray_items[api_name] = 'Method' if isinstance(v, types.FunctionType) else 'Attribute'
+                    else:
+                        deprecated_items.append(k)
+            # elif not module or class and not private
+            elif not key.startswith('_'):
+                if check_not_deprecated(value):
+                    larray_items[key] = 'Function' if isinstance(value, types.FunctionType) else 'Constant'
+                else:
+                    deprecated_items.append(key)
+
+    return larray_items, deprecated_items
+
+
+def get_items_from_api_doc():
+    def add_item(item, api_doc_items):
+        item = item.astext().strip().split()
+        if len(item):
+            if isinstance(item, list):
+                item = item[0]
+            api_doc_items.append(item)
+
+    api_doc_items = []
+    directives.register_directive('autosummary', Autosummary)
+    cleanup_line = lambda line: line.replace(':attr:', '      ').replace('`', ' ')
```

Comment:
I don't really like lambdas... Yes they save one line, but they are less readable. Not worth it IMO.